cluster/assemble.go

```diff
@@ -0,0 +1,120 @@
+package cluster
+
+import (
+	"context"
+	"crypto/ed25519"
+	"crypto/rand"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"errors"
+	"log/slog"
```

Comment:
I forgot to mention this yesterday but Samuele's comment reminded me: this was introduced in a version more recent than what we can use